src/gardenlinux/build/exporter.py

```diff
@@ -0,0 +1,134 @@
+# -*- coding: utf-8 -*-
+
+import os
+import pathlib
+import re
+import shutil
+import subprocess
+from os import PathLike
+
+from elftools.common.exceptions import ELFError
+from elftools.elf.elffile import ELFFile
+
+from ..logger import LoggerSetup
+
+# Parses dependencies from ld output
+parse_output = re.compile("(?:.*=>)?\\s*(/\\S*).*\n")
+# Remove leading /
+remove_root = re.compile("^/")
+
+
+# Check for ELF header
+def _isElf(path: str | PathLike[str]) -> bool:
+    """
+    Checks if a file is an ELF by looking for the ELF header.
+
+    :param path:    Path to file
+
+    :return: (bool) If the file found at path is an ELF
+    :since:  1.0.0
+    """
+
+    with open(path, "rb") as f:
+        try:
+            ELFFile(f)
+            return True
+        except ELFError:
+            return False
+
+
+def _getInterpreter(path: str | PathLike[str]) -> pathlib.Path:
+    """
+    Returns the interpreter of an ELF. Supported architectures: x86_64, aarch64, i686.
+
+    :param path:    Path to file
+
+    :return: (str) Path of the interpreter
+    :since:  1.0.0
+    """
+
+    with open(path, "rb") as f:
+        elf = ELFFile(f)
+        interp = elf.get_section_by_name(".interp")
+
+        if interp:
+            return pathlib.Path(interp.data().split(b"\x00")[0].decode())
+        else:
+            match elf.header["e_machine"]:
+                case "EM_AARCH64":
+                    return pathlib.Path("/lib/ld-linux-aarch64.so.1")
+                case "EM_386":
+                    return pathlib.Path("/lib/ld-linux.so.2")
+                case "EM_X86_64":
+                    return pathlib.Path("/lib64/ld-linux-x86-64.so.2")
+                case arch:
+                    raise RuntimeError(f"Error: Unsupported architecture for {path}: only support x86_64 (003e), aarch64 (00b7) and i686 (0003), but was {arch}")
+                
+def _get_default_package_dir() -> pathlib.Path | None:
+    """
+    Finds the default site-packages or dist-packages directory of the default python3 environment
+
+    :return: (str) Path to directory
+    :since:  1.0.0
+    """
+    
+    # Needs to escape the virtual environment python-gardenlinx-lib is running in
+    interpreter = shutil.which("python3")
+    if interpreter:
```

Comment:
Switching blocks here:
```
if not interpreter:
  raise RuntimeError("...")

out = ...
```

Would result in the return value always returning a `Path`.